presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -159,8 +169,8 @@ public NodeSelector createNodeSelector(ConnectorId connectorId)
                 }
             }
 
-            return new NodeMap(byHostAndPort.build(), byHost.build(), workersByNetworkPath.build(), coordinatorNodeIds);
-        }, 5, TimeUnit.SECONDS);
+            return new NodeMap(byNodeId.build(), byHostAndPort.build(), byHost.build(), workersByNetworkPath.build(), coordinatorNodeIds);
+        }, nodeMapRefreshInNanos, TimeUnit.NANOSECONDS);
```

Comment:
The `memoizeWithExpiration` supplier doesn't allow 0 =\

I recommend to have a condition here. If the duration is 0, then create a regular supplier. Otherwise the `memoizeWithExpiration` one.